src/librustdoc/html/render/print_item.rs

```diff
@@ -1238,6 +1238,72 @@ fn item_typedef(w: &mut Buffer, cx: &mut Context<'_>, it: &clean::Item, t: &clea
 
     write!(w, "{}", document(cx, it, None, HeadingOffset::H2));
 
+    // Only show inner variants if:
+    //  - the typedef does NOT have any generics (modulo lifetimes)
+    //  - AND the aliased type has some generics
+    let should_print_inner_variants = t
+        .generics
+        .params
+        .iter()
+        .all(|param| matches!(param.kind, clean::GenericParamDefKind::Lifetime { .. }))
+        && t.type_.generics().map(|generics| !generics.is_empty()).unwrap_or(false);
+
+    if should_print_inner_variants {
+        fn toggle_open(mut w: impl fmt::Write) {
```

Comment:
Rather than this, could you instead create a `toggle_wrapper` which will call this write, then call the closure to generate the "inner code" then call `toggle_close` at the end? Like that, impossible to forget a close and simpler to follow.